### ISSUE TITLE
Support ingress.kubernetes.io/rewrite-target

### ISSUE THREAD

**Issue by abh** [pomerium/pomerium#2979]
I converted an old Ingress object to use Pomerium and it appears `ingress.kubernetes.io/rewrite-target` isn't supported?

I had a configuration of `ingress.kubernetes.io/rewrite-target: "/"` to make the prefixes configured in the ingress object be stripped, for example

```
      paths:
      - backend:
          service:
            name: someapp
            port:
              number: 80
        path: /someapp
        pathType: ImplementationSpecific
```

will have requests for `pomerium.example.com/someapp/foo` be sent to `someapp.ns.svc.cluster.local/foo` (no path).

I thought first that I could use `host_path_regex_rewrite_pattern` for this, but that's all about rewriting the host header (only).

**Comment by travisgroth** [comment]
Hi @abh would [`prefix_rewrite`](https://www.pomerium.com/reference/#prefix-rewrite) or [`regex_rewrite`](https://www.pomerium.com/reference/#regex-rewrite) do the trick?  I don't think `ingress.kubernetes.io/rewrite-target` is  a standard annotation.

**Comment by unsubtleguy** [comment]
Hey - Not OP, but I can +1 a benefit from `prefix_rewrite` annotation (which doesn't appear to exist in doco).

Also in the same situation of moving from the old policy config blob to individual ingresses :)

**Comment by WojoInc** [comment]
Can +1 this as well. Trying to move to using the Pomerium ingress controller from Traefik + forward auth.

However, I have a few applications which don't support a base path, and as a result I have to rewrite the path during forwarding.
I've tried adding `ingress.pomerium.io/prefix_rewrite: '/'` (though it isn't listed [here](https://www.pomerium.com/docs/k8s/ingress.html#supported-annotations)). I've also tried applying `prefix_rewrite` as part of a policy block with the `ingress.pomerium.io/policy` annotation. 

Either option leads to error events on the ingress:
![image](https://user-images.githubusercontent.com/4422163/164910363-30f73142-2cca-4e72-82f3-4ae358fb3d3c.png)
